File: VB-Examples/07_Conversion/SetGlobalCustomFonts.vb
Imports Spire.Xls

Namespace SetGlobalCustomFonts
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
			' Set custom font directory
			Dim fontPath() As String = { "..\..\..\..\..\..\Data\fonts" }

			' Create a new workbook object
			Dim workbook As New Workbook()

			Workbook.SetGlobalCustomFontsFolders(fontPath)

			' Load an existing Excel file from the specified path
			workbook.LoadFromFile("..\..\..\..\..\..\Data\SpecialFont.xlsx")

			' Save the workbook to PDF 
			Dim result As String = "output.pdf"
			workbook.SaveToFile(result, FileFormat.PDF)

			' Dispose of the workbook object
			workbook.Dispose()

			' View the document using a file viewer
			FileViewer(result)

			Me.Close()
		End Sub

		Private Sub FileViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace

File: CS-Examples/07_Conversion/SetGlobalCustomFonts.cs
using Spire.Xls;
using System;
using System.Windows.Forms;

namespace SetGlobalCustomFonts
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            // Set custom font directory
            string[] fontPath = { @"..\..\..\..\..\..\Data\fonts" };
      

            // Create a new workbook object
            Workbook workbook = new Workbook();
	    Workbook.SetGlobalCustomFontsFolders(fontPath);

            // Load an existing Excel file from the specified path
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\SpecialFont.xlsx");

            // Save the workbook to PDF 
            String result = "output.pdf";
            workbook.SaveToFile(result, FileFormat.PDF);

            // Dispose of the workbook object
            workbook.Dispose();

            // View the document using a file viewer
            FileViewer(result);

            this.Close();
        }

        private void FileViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
    }
}
